# Phase 4 — Exploratory Data Analysis (EDA)

In [3]:
import pandas as pd
df=pd.read_csv("cleaned_data_for_EDA.csv")
df.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit,Order Month,Order Year
0,1,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136,2016-11,2016
1,2,CA-2016-152156,2016-11-08,2016-11-11,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820,2016-11,2016
2,3,CA-2016-138688,2016-06-12,2016-06-16,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714,2016-06,2016
3,4,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310,2015-10,2015
4,5,US-2015-108966,2015-10-11,2015-10-18,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164,2015-10,2015


In [4]:
# Step 1 — Calculate Core Business Metrics
total_sales = df["Sales"].sum()
total_profit = df["Profit"].sum()
total_orders = df["Order ID"].nunique()

print("Total Sales:", total_sales)
print("Total Profit:", total_profit)
print("Total Orders:", total_orders)

Total Sales: 2297200.8603000003
Total Profit: 286397.0217
Total Orders: 5009


## Executive Summary: Overall Business Performance
Based on our initial aggregation, the Superstore's high-level performance metrics are as follows:

| Metric | Value |
| :--- | :--- |
| Total Revenue (Sales) | 2,297,201 |
| Total Net Profit | 286,397 |
| Total Unique Orders | 5,009 |

### Initial Observations:
1. **Scale:** The business is operating at a multi-million dollar scale across a healthy volume of over 5,000 orders.
2. **Overall Profitability:** The business is profitable overall.
3. **Average Revenue per Order:** By dividing Total Sales by Total Orders, we can see the average order brings in roughly **458.61**.

---

## The "Efficiency" Metric: Profit Margin
Revenue (Sales) tells us how much money is coming through the door, but **Profit Margin** tells us how much of that money we actually get to keep. 

### Why Profit Margin Matters
A product can have high sales but still be a "money pit" if its margins are low or negative. By calculating the margin, we can compare a \10,000 category to a \1,000 category on a level playing field.

### The Formula
We calculate the margin for every transaction using the following logic:

$$\text{Profit Margin \%} = \left( \frac{\text{Profit}}{\text{Sales}} \right) \times 100$$

---

In [5]:
profit_margin = (total_profit / total_sales) * 100
print("Profit Margin: ", round(profit_margin, 2),"%")

Profit Margin:  12.47 %


##  Business Interpretation: The 12.47% Benchmark
Our calculation shows that for every 100 in sales, the company keeps **12.47** in profit. 

### Why we need to dig deeper:
While a ~12.5% margin is a solid industry standard for retail, it represents an **aggregate average**. In a diverse business like this, "The Average" can be a dangerous mask. 

**Our next objective is to "unmask" the data by looking for:**
* **Profit Engines:** Categories that are highly efficient and deserve more investment.
* **Drain on Resources:** Categories where we are moving high volume but keeping very little (or losing) money.

---



##  Phase 2: Category Performance
We will now break down **Sales**, **Profit**, and **Profit Margin** by Product Category to see which parts of the catalog are driving the business forward.

business question:

Which products, regions, and customer segments drive revenue?

The first breakdown we analyze is Product Category.

In [6]:
category_sales = df.groupby("Category")["Sales"].sum().sort_values(ascending=False)
category_profit = df.groupby("Category")["Profit"].sum().sort_values(ascending=False)

print(category_sales)
print()
print(category_profit)

Category
Technology         836154.0330
Furniture          741999.7953
Office Supplies    719047.0320
Name: Sales, dtype: float64

Category
Technology         145454.9481
Office Supplies    122490.8008
Furniture           18451.2728
Name: Profit, dtype: float64


##  Category Performance: The Profitability Gap
We have broken down the high-level performance by Category. While the business is healthy overall, a major discrepancy emerges when comparing Revenue to Net Profit.

### 1. Sales vs. Profit Comparison
| Category | Total Sales | Total Profit | Profit Margin % |
| :--- | :--- | :--- | :--- |
| **Technology** | 836,154 | 145,454 | ~17.4% |
| **Office Supplies** | 719,047 | 122,490 | ~17.0% |
| **Furniture** | 741,999 | 18,451 | **~2.5%** |

###  Key Observation: The Furniture Trap
**Furniture** accounts for roughly **32% of total revenue**, yet it contributes only **6% of total profit.** This low margin of **2.5%** indicates that for every dollar of furniture sold, the company barely breaks even. This is a critical risk area for the business. 



### Potential Root Causes to Investigate:
* **Sub-Category Losses:** Is one specific item (like Tables or Chairs) dragging down the whole category?
* **Discounting:** Are we being too aggressive with promotions to move bulky inventory?
* **Regional Variance:** Is Furniture only a problem in specific parts of the country?

---

# Next Step — Sub-Category Analysis

Categories are too broad.
We must analyze Sub-Category performance.

In [7]:
subcat_profit = df.groupby("Sub-Category")[["Sales","Profit"]].sum().sort_values(by="Profit", ascending=False)

subcat_profit

,Sales,Profit
Sub-Category,,
Copiers,149528.0300,55617.8249
Phones,330007.0540,44515.7306
Accessories,167380.3180,41936.6357
Paper,78479.2060,34053.5693
Binders,203412.7330,30221.7633
Chairs,328449.1030,26590.1663
Storage,223843.6080,21278.8264
Appliances,107532.1610,18138.0054
Furnishings,91705.1640,13059.1436


##  Sub-Category Analysis: Identifying the "Profit Killers"
Aggregate category data told us Furniture was struggling; drilling down into Sub-Categories reveals exactly where the "bleeding" is occurring.

### 1. The Stars vs. The Dogs
| Sub-Category | Sales | Profit | Status |
| :--- | :--- | :--- | :--- |
| **Copiers** | 149,528 | **55,617** |  **High-Margin Star:** High profit despite lower volume. |
| **Tables** | 206,965 | **-17,725** |  **Profit Killer:** High volume but actively losing money. |

### 2. Key Business Insights
* **The Copier Advantage:** Copiers are the business’s most efficient engine. We should investigate why their margin is so high and if we can replicate this success in other tech sub-categories.
* **The Tables Crisis:** We sold over **200,000** worth of tables and actually *lost* over **17,000** in the process. Every table sold is currently a liability to the company’s bottom line.
* **The Efficiency Gap:** Items like **Chairs** and **Machines** are moving massive amounts of inventory but keeping very little profit, suggesting they are "busy but not profitable."



###  The Next Logical Question:
Why are Tables losing money? Is it a inherent cost problem, or are we **over-discounting** them to compete in a crowded market?

---

# Next Step — Investigate Discounts

Remember earlier we saw discounts up to 80%.

A common problem in retail datasets is:

High discounts → negative profit.

In [8]:
df[df["Sub-Category"] == "Tables"][["Sales","Discount","Profit"]].describe()

,Sales,Discount,Profit
count,319.000000,319.000000,319.000000
mean,648.794771,0.261285,-55.565771
std,615.774655,0.169622,230.565006
min,24.368000,0.000000,-1862.312400
25%,244.006000,0.200000,-122.829300
50%,447.840000,0.300000,-31.372200
75%,872.170000,0.400000,33.851700
max,4297.644000,0.500000,629.010000


##  Deep Dive: The "Table" Profitability Crisis
We isolated the **Tables** sub-category (319 orders) to understand why it is the primary drain on company resources.

### 1. Statistical Breakdown
| Metric | Value | Business Meaning |
| :--- | :--- | :--- |
| **Average Sales** | 648.79 | Tables are high-ticket items. |
| **Average Discount** | **26%** | Significant price slashing is the norm here. |
| **Average Profit** | **-55.57** | On average, the company loses money every time a table is sold. |
| **Profit Range** | -1,862 to +629 | High volatility; the losses are far more extreme than the gains. |

### 2. The "Discount Trap" Identified
The data shows a clear correlation: Tables sold at a **20% discount or higher** almost never reach profitability. While high discounts drive the **206K in sales volume**, they are simultaneously "destroying" the net profit.



###  Actionable Business Recommendations:
* **Discount Hard-Cap:** Implement an immediate freeze on table discounts exceeding 15%.
* **Product Audit:** Investigate the specific models resulting in the **-1,862** loss—these may be shipping-damaged units or priced incorrectly.
* **Bundle Strategy:** Transition Tables from "standalone" sales to bundles with high-margin items like **Copiers** or **Chairs** to offset the margin loss.

---

# Next Step — Regional Analysis
We’ve covered Products and Sub-Categories. Now we move to regions, another key part of the business question:

Which regions drive revenue and profit?

In [9]:
region_sales = df.groupby("Region")[["Sales","Profit"]].sum().sort_values(by="Sales", ascending=False)
region_sales

,Sales,Profit
Region,,
West,725457.8245,108418.4489
East,678781.2400,91522.7800
Central,501239.8908,39706.3625
South,391721.9050,46749.4303


##  Regional Performance: The Efficiency Gap
While the **West** leads in sheer volume, a comparison between the **Central** and **South** regions reveals a significant disparity in operational efficiency.

### 1. Regional Comparison Table
| Region | Total Sales | Total Profit | Profit Margin % |
| :--- | :--- | :--- | :--- |
| **West** | 725,458 | 108,418 | ~14.9% |
| **South** | 391,722 | 46,749 | **~11.9%** |
| **East** | 678,781 | 91,523 | ~13.5% |
| **Central** | 501,240 | 39,706 | **~7.9%** |

### 2. Key Business Insights
* **The West is the "Gold Standard":** Leading in both total revenue and margin, the West represents the most successful regional model. Best practices from this region should be documented and shared.
* **The Central Region Paradox:** Despite generating over half a million dollars in sales, the Central region has the lowest profit margin (7.9%).
* **South vs. Central Efficiency:** The South is essentially "doing more with less." Even with lower market share, its lean operations result in higher net profit than the Central region.



###  Hypothesis to Test:
Is the **Central** region's low profit caused by the "Table Crisis" we found earlier? Perhaps the Central region is where the most aggressive discounting on Furniture is happening.

---

# Next Step — Customer Segment Analysis

We’ve covered products and regions. Now we analyze customer segments:

Which customer segment generates the most sales

Which segment generates the most profit

Are there segments with high sales but low profit, similar to Tables or Central region?

In [10]:
segment_sales = df.groupby("Segment")[["Sales","Profit"]].sum().sort_values(by="Sales", ascending=False)
segment_sales

,Sales,Profit
Segment,,
Consumer,1.161401e+06,134119.2092
Corporate,7.061464e+05,91979.1340
Home Office,4.296531e+05,60298.6785


##  Customer Segment: Volume vs. Efficiency
Our final level of analysis looks at *who* is buying. We compared the three primary segments: **Consumer**, **Corporate**, and **Home Office**.

### 1. Segment Performance Table
| Segment | Total Sales | Total Profit | Profit Margin % |
| :--- | :--- | :--- | :--- |
| **Consumer** | 1,161,401 | 134,119 | ~11.5% |
| **Corporate** | 706,146 | 91,979 | ~13.0% |
| **Home Office** | 429,653 | 60,299 | **~14.0%** |

### 2. Key Business Insights
* **The Revenue Engine (Consumer):** This segment drives over 50% of total revenue. However, it also has the lowest profit margin, likely due to heavy price sensitivity and higher rates of discounting.
* **The Hidden Gem (Home Office):** While the smallest in total volume, this segment is the most profitable per dollar spent. This suggests that Home Office customers value quality or specific features over the lowest price.
* **Strategic Pivot:** Instead of just chasing more Consumer sales, the business could see a higher Return on Investment (ROI) by targeting the Home Office segment with specialized marketing.



---

##  Phase 3 Summary: Exploratory Data Analysis (EDA)
We have successfully "unmasked" the aggregate data to find the truth:
1. **Product:** **Copiers** are stars; **Tables** are a liability.
2. **Geography:** The **West** is flourishing; the **Central** region is struggling with margins.
3. **Customer:** **Consumers** bring the crowd; **Home Office** brings the profit.

In [11]:
df.to_csv("cleaned_data_for_EDA_visualization.csv", index=False)
print("Data saved to cleaned_data_for_EDA_visualization.csv")

Data saved to cleaned_data_for_EDA_visualization.csv
